### DISCUSSION TITLE
Add preset to color picker

### DISCUSSION THREAD

**Discussion by scoch5** [craftcms/cms#14463] [Ideas] [answered]
HI,
It would be very useful to have the ability to add presets to the "Color" field. Since you are using the standard HTML input type="color" you would just need to add the "list" attribute and a "datalist" element with the presets.

Thanks

**Comment by brandonkelly** [comment, marked as answer]
Easy enough! Just added for Craft 4.8 (e997d90f4aa17c3bf0d7c105e7c95391ed07fc5c).

Looks like Firefox doesn’t support it yet for macOS or Android (https://bugzilla.mozilla.org/show_bug.cgi?id=1805397), but otherwise seems pretty well-supported.

**Reply by brandonkelly** [reply]
Craft 4.8.0 is out now with that addition :)

**Comment by andreimoment** [comment]
Thank you for merging this!

Until now, I have been using a color plugin just for this feature and would love to switch that field to a standard color field.

What would be a good way to migrate data from the plugin back to the standard color field?

**Reply by brandonkelly** [reply]
If the other field type also stores its values as hex colors starting with `#`, you could just switch the field type to Color.

Otherwise, you would need to create a separate Color field, and [bulk-update your entries](https://craftcms.com/knowledge-base/bulk-resaving-elements#resaving-with-computed-field-values) using the `resave/entries` command. Something like:

```sh
php craft resave/entries --set newField --to "fn(\$entry) => sprintf('#%s', \$entry->oldField)"
```

**Comment by scoch5** [comment]
Thanks @brandonkelly, great work!

A further improvement would be to have the list of presets at site or craft installation level to avoid repeating the same presets for each field. Perhaps it could be a configuration to be inserted in the config/general.php so that every single field can use the default one or have one just for itself.
It would become very useful in case matrix blocks have background color or text color configurations and help editors maintain a consistent site palette.

**Reply by brandonkelly** [reply]
Can you post that as a new discussion?
